package_modules/yum

```diff
@@ -0,0 +1,216 @@
+#!/usr/bin/python
+
+import sys
+import os
+import subprocess
+import re
+
+rpm_cmd = "/bin/rpm"
+rpm_output_format = "Name=%{name}\nVersion=%{version}\nArchitecture=%{arch}\n"
+
+yum_cmd = "/usr/bin/yum"
+yum_options = "--quiet"
+
+if (os.environ.get('CFENGINE_TEST_RPM_CMD')):
+    rpm_cmd = os.environ['CFENGINE_TEST_RPM_CMD']
+if (os.environ.get('CFENGINE_TEST_YUM_CMD')):
+    yum_cmd = os.environ['CFENGINE_TEST_YUM_CMD']
+
+def get_package_data():
+    pkg_string = ""
+    for line in sys.stdin:
+        result = re.match("^File=(.*)", line)
+        if (result):
+            pkg_string = result.group(1)
+            break
+
+    if (pkg_string == ""):
+        return 1
+
+    repo = False
+    if (re.match("^/.*", pkg_string)):
+        # Absolute file.
+        sys.stdout.write("PackageType=file\n")
+    elif (re.search("([:,]|-[0-9])", pkg_string)):
+        # Contains either a version number or an illegal symbol.
+        sys.stdout.write(line + "ErrorMessage: Package string with illegal format\n")
+        return 1
+    else:
+        repo = True
+        sys.stdout.write("PackageType=repo\n")
+
+    if (repo):
+        sys.stdout.write("Name=" + pkg_string + "\n")
+        return 0
+    else:
+        sys.stdout.flush()
+        return subprocess.call([rpm_cmd, "--qf", rpm_output_format, "-qp", result.group(1)])
+
+
+def list_installed():
+    for line in sys.stdin:
+        # Ignore everything.
+        pass
+
+    return subprocess.call([rpm_cmd, "-qa", "--qf", rpm_output_format])
+
+
+def list_updates(online):
+    for line in sys.stdin:
+        # Ignore everything.
+        pass
+
+    if online:
+        online_flag = []
+    else:
+        online_flag = ["-C"]
+
+    process = subprocess.Popen([yum_cmd, yum_options] + online_flag + ["check-update"], stdout=subprocess.PIPE)
+    lastline = ""
+    for line in process.stdout:
+        # Combine multiline entries into one line. A line without at least three
+        # space separated fields gets combined with the next line, if that line
+        # starts with a space.
+        if len(lastline) > 0 and line[0] != " ":
```

Comment:
Do you really mean `line[0] != ' '` (only the simple space character) or would `not line[0].isspace()` be more apt ?

Also, `len(lastline) > 0` is better phrased as simply `lastline`, using it as a boolean with less effort and the same effect.